app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -0,0 +1,465 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.editorial.controllers
+
+import play.api.Logger
+import play.api.data.{ Form, FormError }
+import play.api.i18n.{ I18nSupport, Lang }
+import play.api.mvc._
+import play.twirl.api.HtmlFormat
+import uk.gov.nationalarchives.omega.editorial.controllers.EditSetRecordController._
+import uk.gov.nationalarchives.omega.editorial.controllers.authentication.Secured
+import uk.gov.nationalarchives.omega.editorial.forms.EditSetRecordFormValues.{ modifyEditSetRecordWithFormValues, populateForm }
+import uk.gov.nationalarchives.omega.editorial.forms.{ EditSetRecordFormValues, EditSetRecordFormValuesFormProvider }
+import uk.gov.nationalarchives.omega.editorial.models._
+import uk.gov.nationalarchives.omega.editorial.services.CoveringDateCalculator.getStartAndEndDates
+import uk.gov.nationalarchives.omega.editorial.services.{ CoveringDateError, EditSetRecordService, ReferenceDataService }
+import uk.gov.nationalarchives.omega.editorial.support.{ DateParser, FormSupport }
+import uk.gov.nationalarchives.omega.editorial.views.html.{ editSetRecordEdit, editSetRecordEditDiscard, editSetRecordEditSave }
+import uk.gov.nationalarchives.omega.editorial.{ controllers, editSetRecords, editSets }
+
+import java.time.LocalDate
+import java.time.temporal.ChronoField.{ DAY_OF_MONTH, MONTH_OF_YEAR, YEAR }
+import javax.inject.{ Inject, Singleton }
+
+@Singleton
+class EditSetRecordController @Inject() (
+  messagesControllerComponents: MessagesControllerComponents,
+  referenceDataService: ReferenceDataService,
+  editRecordSetService: EditSetRecordService,
+  editSetRecordEdit: editSetRecordEdit,
+  editSetRecordEditDiscard: editSetRecordEditDiscard,
+  editSetRecordEditSave: editSetRecordEditSave
+) extends MessagesAbstractController(messagesControllerComponents) with I18nSupport with Secured with FormSupport {
+
+  private val logger: Logger = Logger(this.getClass)
+  private lazy val legalStatuses: Seq[LegalStatus] = referenceDataService.getLegalStatuses
+  private lazy val creators: Seq[Creator] = referenceDataService.getCreators
+  private lazy val placesOfDeposit: Seq[PlaceOfDeposit] = referenceDataService.getPlacesOfDeposit
+
+  private type RecordTransformer = EditSetRecord => EditSetRecord
```

Comment:
I think that these type should go into the companion object

Reply:
Good idea; but if other classes will use them, maybe they shouldn't belong to this companion object?

Maybe `FormSupport`?

Reply:
Actually, I think `RecordTransformer` would be more at home in the `EditSetRecord` companion object.

Reply:
Changed.